### PR TITLE
Migrate EfficentNetV2 to Backbone

### LINKED ISSUE #1626: Migrate EfficientNetv2 to the new Backbone API
To align KerasCV with the KerasNLP API, we're migrating our backbone models to subclass keras_cv.models.Backbone.
This allows us to have a more structured, modular API for high-level task models, including the new ImageClassifier model (see #1458).

A great step-by-step guide to this migration can be found in the description of #1508, which performed this migration for ResNetV1.

If you'd like to take this issue up, please leave a comment and the KerasCV team can assign the issue to you. When opening a PR, please tag this issue as fixed by your PR, and tag our tracking bug #1627 for bookkeeping.

### PR DESCRIPTION
**Pull request by cosmo3769**
# What does this PR do?

Fixes #1626 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue?
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

@LukeWood @ianstenbit

### CODE REVIEW

**Comment by soma2000-lang** [comment]
@cosmo3769 I have already made a pr on EfficientV2.Can you please make sure you the same work is not duplicated

File: keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    This every presets for EfficientNetV2 and is only run manually.
+    Run with:
+    `pytest keras_cv/models/backbones/efficientnet_v2/
+    efficientnet_v2_backbone_presets_test.py
+    --run_extra_large`
+    """
+
+    def test_load_regnet(self):
+        input_data = tf.ones(shape=(2, 224, 224, 3))
+        for preset in EfficientNetV2Backbone.presetsx:
```
Comment (LukeWood) [inline keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py:38]: looks like `presetsx` is a typo - does this pass?

File: keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            "activation": "swish",
+            "blocks_args": "default",
+            "include_rescaling": True,
+            "input_shape": (None, None, 3),
+            "input_tensor": None,
+        },
+    },
+}
+
+backbone_presets_with_weights = {}
```
Comment (LukeWood) [inline keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets.py:167]: Please add presets for all of the efficientnet weights!

https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/weights.py#L81

Comment (LukeWood) [reply keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets.py:167]: https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/weights.py#L163

**Comment by LukeWood** [comment]
Hey @soma2000-lang and @cosmo3769 !  I'm going to take this PR over as theres a lot of changes that need to be made and this is a very high priority for the team and its blocking a few threads of work.   

I'll push commits to your branch, so authorship should still show up.  I just wanted to give a heads up!  Feel free to leave this open until merged, and I'll CC you on the PR!

**Comment by cosmo3769** [comment]
Hey @LukeWood! Thank you for letting me know and keeping me in the loop. I understand the urgency and appreciate your heads up. I'll keep an eye out for updates on the PR.

**Comment by soma2000-lang** [comment]
Hi @LukeWood @cosmo3769 Can I push some changes to this PR also ,so that  I still have some authorship, for the PRS I made in   #1718 and #1645 .

**Comment by LukeWood** [comment]
Closing in favor of the latter PR

**Pull request closed without merging**